### PR TITLE
Defer Directory.Exists check for performance reasons when determining…

### PR DESCRIPTION
**Pull request by michaelcfanning**
… file mime type. @ScottLouvau, FYI

### CODE REVIEW

File: src/Sarif/Writers/MimeType.cs [hunk 1]
```diff
@@ -39,6 +34,12 @@ public static string DetermineFromFileExtension(string path)
                     }
                 }
             }
+
+            if (System.IO.Directory.Exists(path))
```
Comment (ScottLouvau) [inline src/Sarif/Writers/MimeType.cs]: Great! This should help quite a bit by itself.

Did you also defer the check in FileData.Create when we didn't request any dataToInsert? That was the other half of the performance issue.

Comment (ScottLouvau) [reply src/Sarif/Writers/MimeType.cs]: Cool, that works. =)

**Comment by michaelcfanning** [comment]
@ScottLouvau did you have a chance to look at this?

**Comment by ScottLouvau** [comment]
Looks good; I reviewed.

**Pull request merged**